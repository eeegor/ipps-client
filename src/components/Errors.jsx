import React from 'react';
import './Errors.scss';
import { guid } from '../util';

const Error = props => <div className="error">{props.error}</div>;

export const Errors = props => {
  const { errors } = props;
  return (
    <div className="errors">
      {errors.map(error => {
        if (error.message) {
          const { response } = error;
          const nestedType =
            (response && response.data && response.data.message) || null;
          const nestedMessage =
            (response && response.data && response.data.error.message) || null;
          if (nestedType) {
            if (nestedMessage) {
              return <Error key={guid()} error={nestedMessage} />;
            }
            return <Error key={guid()} error={nestedType} />;
          }
          switch (error.message) {
            case 'Request failed with status code 400':
              const serverError = 'Server error, please try again';
              return <Error key={guid()} error={serverError} />;
            case 'Request failed with status code 401':
              const notAuthorized =
                'You need to be authenticated to view this page';
              return <Error key={guid()} error={notAuthorized} />;
            case 'Network Error':
              const message = error.message;
              return <Error key={guid()} error={message} />;
            case 'Validation Error':
              return <Error key={guid()} error={error.error} />;
            default:
              const defaultMessage = 'Something went wrong, please try again';
              return <Error key={guid()} error={defaultMessage} />;
          }
        }

        // return <div key={guid()} className="error">{nextError.message}</div>;
        // const name = data.name;
        // const dataErrors = data.errors;
        // const code = data.code;

        // if (message && !dataErrors) {
        //   return <div className="error">{`${message} `}</div>;
        // }

        // if (typeof data === 'string' && data !== '') {
        //   return <div className="error">{`${data} `}</div>;
        // }

        // return (
        //   <div className="error-wrap" key={guid()}>
        //     {name && code && (
        //       <div className="error">
        //         {code !== 11000 && `code: ${code}`}
        //         {code === 11000 && `This user already exists`}
        //       </div>
        //     )}
        //     {dataErrors &&
        //       Object.keys(dataErrors).map(errorName => {
        //         const properties = dataErrors[errorName].properties;
        //         return (
        //           <div className="error" key={errorName}>
        //             {properties.minlength &&
        //               `${properties.path}, min length: ${properties.minlength}`}
        //             {!properties.minlength &&
        //               properties.path &&
        //               `${properties.path} is required`}
        //           </div>
        //         );
        //       })}
        //   </div>
        // );
      })}
    </div>
  );
};
