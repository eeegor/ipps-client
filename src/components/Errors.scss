@import '../variables';

.errors {
  display: flex;
  flex-direction: column;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  background: rgba($color--danger, 0.8);
  color: white;
}

.error-wrap {
  padding: 0.5rem 0;
}

.error {
  padding: 0.5rem 1rem;
}
