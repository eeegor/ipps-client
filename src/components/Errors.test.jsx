import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Errors } from './Errors';

describe('Errors', () => {
  it('renders markup correctly', () => {
    const wrapper = (
      <Errors errors={[]} />
    );
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
