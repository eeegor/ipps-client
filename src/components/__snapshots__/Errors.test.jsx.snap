// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors renders markup correctly 1`] = `
<div
  className="errors"
/>
`;
